src/librustc/ich/fingerprint.rs

```diff
@@ -40,16 +40,16 @@ impl Fingerprint {
 impl Encodable for Fingerprint {
     #[inline]
     fn encode<S: Encoder>(&self, s: &mut S) -> Result<(), S::Error> {
-        s.emit_u64(self.0.to_le())?;
-        s.emit_u64(self.1.to_le())
+        s.emit_u64(self.0)?;
+        s.emit_u64(self.1)
     }
 }
 
 impl Decodable for Fingerprint {
     #[inline]
     fn decode<D: Decoder>(d: &mut D) -> Result<Fingerprint, D::Error> {
-        let _0 = u64::from_le(d.read_u64()?);
-        let _1 = u64::from_le(d.read_u64()?);
+        let _0 = d.read_u64()?;
+        let _1 = d.read_u64()?;
         Ok(Fingerprint(_0, _1))
```

Comment:
Can you move it to use `#[derive]`? Looks like it's what deriving would produce, now.